src/display/process/allProcs.go

```diff
@@ -182,78 +181,76 @@ func AllProcVisuals(dataChannel chan []*proc.Process,
 			case "<Resize>":
 				updateUI() // updateUI only during resize event
 			case "<Escape>":
-				if paused {
+				if killSelected {
 					runAllProc = true
-					paused = false
+					killSelected = false
 					myPage.BodyList.SelectedRowStyle = selectedStyle
 				}
 			case "s": //s to pause
-				if !paused {
+				if !killSelected {
 					pauseProc()
 				}
 			case "j", "<Down>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollDown()
 				}
 			case "k", "<Up>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollUp()
 				}
 			case "<C-d>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollHalfPageDown()
 				}
 			case "<C-u>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollHalfPageUp()
 				}
 			case "<C-f>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollPageDown()
 				}
 			case "<C-b>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollPageUp()
 				}
 			case "g":
-				if !paused && previousKey == "g" {
+				if !killSelected && previousKey == "g" {
 					myPage.BodyList.ScrollTop()
 				}
 			case "<Home>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollTop()
 				}
 			case "G", "<End>":
-				if !paused {
+				if !killSelected {
 					myPage.BodyList.ScrollBottom()
 				}
 			case "K", "<F9>":
-				updateProcs()
-
 				if myPage.BodyList.SelectedRow < len(myPage.BodyList.Rows) {
 					row := myPage.BodyList.Rows[myPage.BodyList.SelectedRow]
 					// get PID from the data
 					pid64, err := strconv.ParseInt(strings.SplitN(row, " ", 2)[0], 10, 32)
 					if err != nil {
 						return fmt.Errorf("Failed to get PID of process: %v", err)
 					}
-					pid := int32(pid64)
+					pidToKill = int32(pid64)
 
-					if !paused {
+					if !killSelected {
 						runAllProc = false
-						paused = true
+						killSelected = true
 						myPage.BodyList.SelectedRowStyle = killingStyle
 					} else {
 						// get process and kill it
-						procToKill, err := proc.NewProcess(pid)
+						procToKill, err := proc.NewProcess(pidToKill)
 						if err == nil {
 							err = procToKill.Kill()
 							if err != nil {
-								return fmt.Errorf("Failed to kill process with PID %d: %v", pid, err)
+								return fmt.Errorf("Failed to kill process with PID %d: %v", pidToKill, err)
 							}
 						}
```

Comment:
@Samyak2 Wouldn't returning this error hinder with the UI? Because in the `proc.Kill()` function, it checks again if the PID exists and then sends a signal. I'm against returning an error here if `proc.kill()` fails as it would unnecessarily stop the UI and quit. I'd drop the check entirely.

Maybe let's wait for @MadhavJivrajani for a better opinion?

Reply:
I'm returning errors in two places (inside kill):
 - If parsing the PID fails: this shouldn't happen in any condition, unless there's a non-numeric PID printed somehow. I think we should keep this since this error will mean that something is seriously wrong.
 - Process kill fails: this could mean permission issues. We could probably ignore this, but there must be some way to let the user know that it failed.

Reply:
@Samyak2 

> but there must be some way to let the user know that it failed.

I'd prefer this than closing the UI. Maybe render a temporary textbox to deliver the message?

Reply:
Yes, I'd also prefer some way of letting the user know and not closing the UI. Maybe instead of the textbox, we could have something simpler like making the color of the selected row as red or something of the sort and we could mention the meaning of doing this as part of #42

Reply:
I thought of the same thing. But I thought it'll hinder experience. Considering the user will have to keep popping the help text open just to understand what's going on. Maybe we could get the row to turn red for now and maybe open another issue to address the delivery of the message?

Reply:
Yes, sounds good. Currently, the row is highlighted in red when it is selected for killing, I'll change the colors.